promptolution/optimizers/opro.py

```diff
@@ -0,0 +1,60 @@
+from typing import List
+
+import numpy as np
+
+from promptolution.llms.base_llm import BaseLLM
+from promptolution.optimizers.base_optimizer import BaseOptimizer
+
+
+class Opro(BaseOptimizer):
+    """
+    Opro: Optimization by PROmpting
+
+    Proposed by the paper "Large Language Models as Optimizers" by Yang et. al: https://arxiv.org/abs/2309.03409
+
+    """
+
+    def __init__(self, llm: BaseLLM, n_samples: int = 2, **args):
```

Comment:
specify what n_samples means (we use this also as n_samples of prompts) our users should not get confused by this